File: components/backend/types/bugfix.go
package types

// BugFix Workflow Data Structures
type BugFixWorkflow struct {
	ID                      string            `json:"id"`
	GithubIssueNumber       int               `json:"githubIssueNumber"`
	GithubIssueURL          string            `json:"githubIssueURL"`
	Title                   string            `json:"title"`
	Description             string            `json:"description"`
	BranchName              string            `json:"branchName"`
	ImplementationRepo      GitRepository     `json:"implementationRepo"` // The repository containing the code/bug
	JiraTaskKey             *string           `json:"jiraTaskKey,omitempty"`
	JiraTaskURL             *string           `json:"jiraTaskURL,omitempty"`
	LastSyncedAt            *string           `json:"lastSyncedAt,omitempty"` // RFC3339 format
	WorkspacePath           string            `json:"workspacePath,omitempty"`
	CreatedBy               string            `json:"createdBy,omitempty"`
	CreatedAt               string            `json:"createdAt,omitempty"`
	UpdatedAt               string            `json:"updatedAt,omitempty"`
	Project                 string            `json:"project,omitempty"`
	Phase                   string            `json:"phase,omitempty"` // Initializing, Ready
	Message                 string            `json:"message,omitempty"`
	AssessmentStatus        string            `json:"assessmentStatus,omitempty"`        // "", "complete"
	ImplementationCompleted bool              `json:"implementationCompleted,omitempty"`
	Annotations             map[string]string `json:"annotations,omitempty"` // Optional metadata
}

// CreateBugFixWorkflowRequest represents the request to create a BugFix Workspace
type CreateBugFixWorkflowRequest struct {
	// Option 1: From GitHub Issue URL
	GithubIssueURL *string `json:"githubIssueURL,omitempty"`

	// Option 2: From text description (creates GitHub Issue automatically)
	TextDescription *TextDescriptionInput `json:"textDescription,omitempty"`

	// Common fields
	ImplementationRepo GitRepository `json:"implementationRepo" binding:"required"` // The repository containing the code/bug
	BranchName         *string       `json:"branchName,omitempty"`                  // Optional, auto-generated if not provided
}

// TextDescriptionInput represents input for creating workspace from text description
type TextDescriptionInput struct {
	Title               string  `json:"title" binding:"required,min=5,max=200"`
	Symptoms            string  `json:"symptoms" binding:"required,min=20"`
	ReproductionSteps   *string `json:"reproductionSteps,omitempty"`
	ExpectedBehavior    *string `json:"expectedBehavior,omitempty"`
	ActualBehavior      *string `json:"actualBehavior,omitempty"`
	AdditionalContext   *string `json:"additionalContext,omitempty"`
	TargetRepository    string  `json:"targetRepository" binding:"required"` // Where to create the GitHub Issue
}

// UpdateBugFixWorkflowRequest represents updates to workspace
type UpdateBugFixWorkflowRequest struct {
	Title        *string `json:"title,omitempty"`
	Description  *string `json:"description,omitempty"`
	JiraTaskKey  *string `json:"jiraTaskKey,omitempty"`
	LastSyncedAt *string `json:"lastSyncedAt,omitempty"`
}

// CreateBugFixSessionRequest represents the request to create a session
type CreateBugFixSessionRequest struct {
	SessionType          string              `json:"sessionType" binding:"required"` // bug-review, bug-resolution-plan, bug-implement-fix
	Prompt               *string             `json:"prompt,omitempty"`               // Custom prompt (auto-generated if not provided)
	Title                *string             `json:"title,omitempty"`
	Description          *string             `json:"description,omitempty"`
	SelectedAgents       []string            `json:"selectedAgents,omitempty"`     // Agent personas
	Interactive          *bool               `json:"interactive,omitempty"`        // Interactive mode using inbox/outbox (default: false/headless)
	AutoPushOnComplete   *bool               `json:"autoPushOnComplete,omitempty"` // Auto-push to GitHub on completion (default: true)
	EnvironmentVariables map[string]string   `json:"environmentVariables,omitempty"`
	ResourceOverrides    *ResourceOverrides  `json:"resourceOverrides,omitempty"`
}

// SyncJiraRequest represents the request to sync workspace to Jira
type SyncJiraRequest struct {
	Force bool `json:"force,omitempty"` // Force sync even if recently synced
}

// SyncJiraResponse represents the response from Jira sync
type SyncJiraResponse struct {
	Success       bool    `json:"success"`
	JiraTaskKey   string  `json:"jiraTaskKey,omitempty"`
	JiraTaskURL   string  `json:"jiraTaskURL,omitempty"`
	Created       bool    `json:"created"` // true if newly created, false if updated
	Message       string  `json:"message,omitempty"`
	LastSyncedAt  string  `json:"lastSyncedAt,omitempty"`
}

// BugFixSession represents a session linked to a BugFix Workspace
type BugFixSession struct {
	ID              string            `json:"id"`
	WorkflowID      string            `json:"workflowId"`
	SessionType     string            `json:"sessionType"` // bug-review, bug-resolution-plan, bug-implement-fix, generic
	Title           string            `json:"title"`
	Description     string            `json:"description,omitempty"`
	Phase           string            `json:"phase"` // Pending, Creating, Running, Completed, Failed, Stopped
	CreatedAt       string            `json:"createdAt"`
	CompletedAt     *string           `json:"completedAt,omitempty"`
	AgentPersonas   []string          `json:"agentPersonas,omitempty"`
}

File: components/frontend/src/services/api/bugfix.ts
/**
 * BugFix Workspace API service
 * Handles all BugFix workflow-related API calls
 */

import { apiClient } from './client';

/**
 * BugFix Workspace types
 */
export interface BugFixWorkflow {
  id: string;
  githubIssueNumber: number;
  githubIssueURL: string;
  title: string;
  description?: string;
  branchName: string;
  phase: 'Initializing' | 'Ready';
  message?: string;
  implementationCompleted?: boolean;
  project: string;
  createdAt: string;
  createdBy: string;
  jiraTaskKey?: string;
  jiraTaskURL?: string;
  lastSyncedAt?: string;
  workspacePath?: string;
  implementationRepo: {
    url: string;
    branch?: string;
  };
}

export interface TextDescriptionInput {
  title: string;
  symptoms: string;
  reproductionSteps?: string;
  expectedBehavior?: string;
  actualBehavior?: string;
  additionalContext?: string;
  targetRepository: string;
}

export interface CreateBugFixWorkflowRequest {
  githubIssueURL?: string;
  textDescription?: TextDescriptionInput;
  implementationRepo: {
    url: string;
    branch?: string;
  };
  branchName?: string;
}

export interface CreateBugFixSessionRequest {
  sessionType: 'bug-review' | 'bug-implement-fix';
  title?: string;
  prompt?: string;
  description?: string;
  selectedAgents?: string[];
  interactive?: boolean;
  autoPushOnComplete?: boolean;
  autoCreatePR?: boolean;
  environmentVariables?: Record<string, string>;
  resourceOverrides?: {
    cpu?: string;
    memory?: string;
    storageClass?: string;
    priorityClass?: string;
    model?: string;
    temperature?: number;
    maxTokens?: number;
    timeout?: number;
  };
}

export interface BugFixWorkflowStatus {
  id: string;
  phase: string;
  message: string;
  implementationCompleted: boolean;
  githubIssueNumber: number;
  githubIssueURL: string;
  jiraSynced: boolean;
  jiraTaskKey?: string;
  lastSyncedAt?: string;
}

export interface BugFixSession {
  id: string;
  title: string;
  sessionType: string;
  phase: string;
  createdAt: string;
  completedAt?: string;
}

export interface SyncJiraRequest {
  force?: boolean;
}

export interface SyncJiraResponse {
  success: boolean;
  jiraTaskKey?: string;
  jiraTaskURL?: string;
  created: boolean;
  message?: string;
  lastSyncedAt?: string;
}

/**
 * List BugFix workflows for a project
 */
export async function listBugFixWorkflows(projectName: string): Promise<BugFixWorkflow[]> {
  const response = await apiClient.get<{ workflows: BugFixWorkflow[] }>(
    `/projects/${projectName}/bugfix-workflows`
  );
  return response.workflows || [];
}

/**
 * Get a single BugFix workflow
 */
export async function getBugFixWorkflow(
  projectName: string,
  workflowId: string
): Promise<BugFixWorkflow> {
  return apiClient.get<BugFixWorkflow>(
    `/projects/${projectName}/bugfix-workflows/${workflowId}`
  );
}

/**
 * Create a new BugFix workflow
 */
export async function createBugFixWorkflow(
  projectName: string,
  data: CreateBugFixWorkflowRequest
): Promise<BugFixWorkflow> {
  return apiClient.post<BugFixWorkflow, CreateBugFixWorkflowRequest>(
    `/projects/${projectName}/bugfix-workflows`,
    data
  );
}

/**
 * Delete a BugFix workflow
 */
export async function deleteBugFixWorkflow(
  projectName: string,
  workflowId: string
): Promise<void> {
  await apiClient.delete(`/projects/${projectName}/bugfix-workflows/${workflowId}`);
}

/**
 * Get BugFix workflow status
 */
export async function getBugFixWorkflowStatus(
  projectName: string,
  workflowId: string
): Promise<BugFixWorkflowStatus> {
  return apiClient.get<BugFixWorkflowStatus>(
    `/projects/${projectName}/bugfix-workflows/${workflowId}/status`
  );
}

/**
 * Create a session for a BugFix workflow
 */
export async function createBugFixSession(
  projectName: string,
  workflowId: string,
  data: CreateBugFixSessionRequest
): Promise<BugFixSession> {
  return apiClient.post<BugFixSession, CreateBugFixSessionRequest>(
    `/projects/${projectName}/bugfix-workflows/${workflowId}/sessions`,
    data
  );
}

/**
 * List sessions for a BugFix workflow
 */
export async function listBugFixSessions(
  projectName: string,
  workflowId: string
): Promise<BugFixSession[]> {
  const response = await apiClient.get<{ sessions: BugFixSession[] }>(
    `/projects/${projectName}/bugfix-workflows/${workflowId}/sessions`
  );
  return response.sessions || [];
}

/**
 * Sync BugFix workflow to Jira
 */
export async function syncBugFixToJira(
  projectName: string,
  workflowId: string,
  data?: SyncJiraRequest
): Promise<SyncJiraResponse> {
  return apiClient.post<SyncJiraResponse, SyncJiraRequest>(
    `/projects/${projectName}/bugfix-workflows/${workflowId}/sync-jira`,
    data || {}
  );
}
